### PR TITLE
Webapp login use components

### PR DESCRIPTION
**Pull request by nfelger**
# Short Description
- This is the final step to start introducing the new frontend: it hooks up the login page using React components!

# Changes
- Changes the render method in `webapp/app/forms/steps/unlock_code_activation_steps.py` to provide all necessary props to the client-side components. No more WTForms Fields in templates!
- Simplifies the UnlockCodeField to perform nothing more than validation and splitting/concatenation.
- Creates a new IdNrField with a similar purpose but keeps the old one around as "Legacy", to be removed once it's no longer used anywhere else.

# Feedback
- Any and all :)

### CODE REVIEW

**Comment by nfelger** [comment]
Side note: this code passes the new acceptance tests.

File: webapp/app/forms/fields.py [hunk 1]
```diff
@@ -162,35 +162,15 @@ def __call__(self, field, **kwargs):
...
-
-    def __call__(self, *args, **kwargs):
-        kwargs = self.set_inputmode(kwargs)
-
-        return super().__call__(*args, **kwargs)
-
-
 class UnlockCodeField(StringField):
-    def __init__(self, label='', validators=None, **kwargs):
-        super(UnlockCodeField, self).__init__(label, validators, **kwargs)
```
Comment (frederike-ramin) [inline webapp/app/forms/fields.py:178]: Can you also make the UnlockCodeField a SteuerlotseStringField?

File: webapp/app/forms/steps/unlock_code_activation_steps.py [hunk 2]
```diff
@@ -34,7 +28,27 @@ def __init__(self, **kwargs):
...
+                        'value': field._value(),
+                        'errors': field.errors,
+                    }
+                    for field in render_info.form
+                },
+            }),
+            # TODO: These are still required by base.html to set the page header.
             form=render_info.form,
             render_info=render_info,
             header_title=self.header_title
```
Comment (frederike-ramin) [inline webapp/app/forms/steps/unlock_code_activation_steps.py:54]: Is that true? I don't think we need the render_info in the base.html if we have a header_title - which we explicitly set here.

**Review by frederike-ramin — COMMENTED** [review]
So great! I like the renaming to LegacyIdNrField - makes it clear to review this. :)
Just a few additional points that I couldn't pinpoint to a line of code:
- [ ] At one point we made sure that all fields are escaping the user's input by using WTForms fields. I'm not sure anymore if it was the fields or the widget that escapes the inputs. But I' sure that you thought about that and we don't introduce a potential security risk?
- [ ] For some reason, the details header links appear black instead of blue on the login page. Maybe some mess-up with the classes?

File: webapp/app/forms/steps/unlock_code_activation_steps.py (hunk 2 above)
Comment (nfelger) [reply webapp/app/forms/steps/unlock_code_activation_steps.py:54]: Good point. I took one look at how the page title is set in `base.html` and balked immediately at what I saw. 😆 I'll revisit this.

Comment (nfelger) [reply webapp/app/forms/steps/unlock_code_activation_steps.py:54]: Maybe I should look at how hard it is to explicitly set the `header_title` everywhere so we can get rid of that ugly logic in the template.

**Review by JulianRoesner — APPROVED** [review]
LGTM

File: webapp/app/forms/steps/unlock_code_activation_steps.py [hunk 3]
```diff
@@ -34,7 +28,27 @@ def __init__(self, **kwargs):
...
+            component='LoginPage',
+            props=humps.camelize({
+                'step_header': {
+                    'title': render_info.step_title,
+                    'intro': render_info.step_intro,
+                },
+                'form': {
+                    'action': render_info.submit_url,
+                    'csrf_token': generate_csrf(),
+                    'show_overview_button': bool(render_info.overview_url),
```
[the remark this answers is not in the document]
Comment (nfelger) [reply webapp/app/forms/steps/unlock_code_activation_steps.py:41]: Where would you put it? The button is inside the `<form>` and submits the form, so it feels natural to me that it's part of the form's properties.

[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/app/forms/steps/unlock_code_activation_steps.py:41]: Yes, makes sense. I have no better idea. I was just a little confused.

**Comment by nfelger** [comment]
@frederike-ramin 
> At one point we made sure that all fields are escaping the user's input by using WTForms fields. I'm not sure anymore if it was the fields or the widget that escapes the inputs. But I' sure that you thought about that and we don't introduce a potential security risk?

I did consider this point, but I didn't test it. React does injection protection by default, so in theory we should be safe (unless we use `dangerouslySetInnerHTML`, which we only do in a single place but we don't render user input that way). I'll test this to be sure though. 👍 

> For some reason, the details header links appear black instead of blue on the login page. Maybe some mess-up with the classes?

Damn, that's some clash with components.css that I didn't anticipate! It happens because I changed the styling too much. I've changed the base.html template to only include components.css on pages that don't set a component. It's a bit of a hack, but since we're migrating page-by-page, this should be fine. Do you see any issues with that?

**Pull request merged**